dapps/valuetransfers/packages/test/tangle_test.go

```diff
@@ -0,0 +1,115 @@
+	"github.com/iotaledger/goshimmer/dapps/valuetransfers/packages/transaction"
+	"github.com/iotaledger/goshimmer/dapps/valuetransfers/packages/wallet"
+)
+
+func TestTangle_ValueTransfer(t *testing.T) {
+	// initialize required objects
+	valueTangle := tangle.New(mapdb.NewMapDB())
+	ledgerState := tangle.NewLedgerState(valueTangle)
+	seed := wallet.NewSeed()
+
+	// setup consensus rules
+	consensus.NewFCOB(valueTangle, 0)
+
+	// check if ledger empty first
+	assert.Equal(t, map[balance.Color]int64{}, ledgerState.Balances(seed.Address(0)))
+	assert.Equal(t, map[balance.Color]int64{}, ledgerState.Balances(seed.Address(1)))
+
+	// load snapshot
+	valueTangle.LoadSnapshot(map[transaction.ID]map[address.Address][]*balance.Balance{
+		transaction.GenesisID: {
+			seed.Address(0): []*balance.Balance{
+				balance.New(balance.ColorIOTA, 337),
+			},
+
+			seed.Address(1): []*balance.Balance{
+				balance.New(balance.ColorIOTA, 1000),
+			},
+		},
+	})
+
+	// check if balance exists after loading snapshot
+	assert.Equal(t, map[balance.Color]int64{balance.ColorIOTA: 337}, ledgerState.Balances(seed.Address(0)))
+	assert.Equal(t, map[balance.Color]int64{balance.ColorIOTA: 1000}, ledgerState.Balances(seed.Address(1)))
+
+	// introduce logic to record liked payloads
+	recordedLikedPayloads, resetRecordedLikedPayloads := recordLikedPayloads(valueTangle)
+
+	// attach first spend
+	outputAddress1 := address.Random()
+	attachedPayload1 := payload.New(payload.GenesisID, payload.GenesisID, transaction.New(
+		transaction.NewInputs(
+			transaction.NewOutputID(seed.Address(0), transaction.GenesisID),
+			transaction.NewOutputID(seed.Address(1), transaction.GenesisID),
+		),
+
+		transaction.NewOutputs(map[address.Address][]*balance.Balance{
+			outputAddress1: {
+				balance.New(balance.ColorIOTA, 1337),
+			},
+		}),
+	))
+	valueTangle.AttachPayloadSync(attachedPayload1)
+
+	// check if old addresses are empty and new addresses are filled
+	assert.Equal(t, map[balance.Color]int64{}, ledgerState.Balances(seed.Address(0)))
+	assert.Equal(t, map[balance.Color]int64{}, ledgerState.Balances(seed.Address(1)))
+	assert.Equal(t, map[balance.Color]int64{balance.ColorIOTA: 1337}, ledgerState.Balances(outputAddress1))
+	assert.Equal(t, 1, len(recordedLikedPayloads))
+	_, payloadLiked := recordedLikedPayloads[attachedPayload1.ID()]
+	assert.True(t, payloadLiked)
+
+	resetRecordedLikedPayloads()
+
+	// attach double spend
+	outputAddress2 := address.Random()
+	valueTangle.AttachPayloadSync(payload.New(payload.GenesisID, payload.GenesisID, transaction.New(
+		transaction.NewInputs(
+			transaction.NewOutputID(seed.Address(0), transaction.GenesisID),
+			transaction.NewOutputID(seed.Address(1), transaction.GenesisID),
+		),
+
+		transaction.NewOutputs(map[address.Address][]*balance.Balance{
+			outputAddress2: {
+				balance.New(balance.ColorNew, 1337),
+			},
+		}),
+	)))
+
+	// shutdown tangle
+	valueTangle.Shutdown()
```

Comment:
How about defering this right after the New to keep construction and destruction closer to each other?